lib/tasks/evm.rake

```diff
@@ -55,6 +55,19 @@ namespace :evm do
     EvmApplication.update_stop
   end
 
+  desc "Determine the deployment scenario"
+  task :deployment_status => :environment do
+    status_to_code = {
+      "new_deployment" => 0,
+      "new_replica"    => 1,
+      "redeployment"   => 2,
+      "upgrade"        => 3
```

Comment:
I gave no thought to this at all.

The previous implementation was setting an environment variable to a string, but I figure an exit code was cleaner so I'm up for suggestions or I'll just go with 3, 4, 5, 6 (based on @Fryguy 's link)